source/Dgraph/Channels/CloudChannel.cs

```diff
@@ -16,30 +16,63 @@
 
 using Grpc.Core;
 using Grpc.Net.Client;
-using System.Threading.Tasks;
 
 namespace Dgraph
 {
     public static class DgraphCloudChannel
     {
+        private const string CloudPort = "443";
+
+        /// <summary>
+        /// Create a new TLS connection to a Dgraph Cloud backend.
+        /// </summary>
+        /// <returns>A new instance of <see cref="GrpcChannel"/></returns>
+        /// <exception cref="ArgumentException"></exception>
         public static GrpcChannel Create(string address, string apiKey)
         {
-            var credentials = CallCredentials.FromInterceptor((context, metadata) =>
+            if (string.IsNullOrWhiteSpace(address))
```

Comment:
I think in this whole part my way of doing this is better check out this PR
https://github.com/dgraph-io/dgraph.net/pull/30/files#diff-c916fd418503f05c72a86508f61f4fe4aaeb5a852677425499568caa6961d660R51

We start already invalidating any address that is not from the cloud. And I do some more checks after that.

Reply:
BTW, I also added some tests.

Reply:
Looks good, I replaced this with your implementation & added the tests in #34

Reply:
We could just merge that PR that is waiting for ages... anyway.

Reply:
@sachyco are you on our slack? Have you checked Michael's PR? can you get something useful from there? Otherwise I will close that one.

Reply:
@MichelDiz No I haven't been invited to the slack. 

Lets keep that PR open for now. Fixing Login & JWT support is out of scope for our current work agreement, but I think it will be very useful when we come to that phase.